File: day1/main.c
#include <stdlib.h>
#include <stdio.h>
#include <fcntl.h>    //open
#include <unistd.h>   //read, write, close
#include <errno.h>    //errno
#include <sys/stat.h> //umask, stat

int sub(int fuel) { return (fuel < 0) ? 0 : sub(fuel / 3 - 2) + fuel; }

int solve(int part) {
    int SIZE = 100;
    int masses[SIZE];
    FILE * file = fopen("input.txt", "r");

    int i;
    int total = 0;
    for (i = 0; i < SIZE; i++) {
        fscanf(file, "%d", &masses[i]);
        int fuel = masses[i] / 3 - 2;
        total += (part == 1) ? fuel : sub(fuel);
    }

    fclose(file);

    return total;
}

int main() {
    printf("Answer to part 1: %d\n", solve(1));
    printf("Answer to part 2: %d\n", solve(2));
    return 0;
}

File: day1/main.py
def main(part):
    masses = open("input.txt").read().split('\n')
    fuels = [int(mass) // 3 - 2 for mass in masses]
    total = 0
    for fuel in fuels:
        if part == 1:
            total += fuel
        else:
            total += sub(fuel)
    return total


def sub(fuel):
    if(fuel < 0):
        return 0
    else:
        return sub(fuel // 3 - 2) + fuel


if __name__ == "__main__":
    print("Answer to part 1:", main(1))
    print("Answer to part 2:", main(2))
